pure.zsh

```diff
@@ -253,19 +266,22 @@ prompt_pure_async_vcs_info() {
 	# to be used or configured as the user pleases.
 	zstyle ':vcs_info:*' enable git
 	zstyle ':vcs_info:*' use-simple true
-	# Only export three message variables from `vcs_info`.
-	zstyle ':vcs_info:*' max-exports 3
-	# Export branch (%b), Git toplevel (%R), and action (rebase/cherry-pick) (%a).
-	zstyle ':vcs_info:git*' formats '%b' '%R'
-	zstyle ':vcs_info:git*' actionformats '%b' '%R' '%a'
+	# Only export four message variables from `vcs_info`.
+	zstyle ':vcs_info:*' max-exports 4
+	# Export branch (%b), Git toplevel (%R), action (rebase/cherry-pick) (%a),
+	# and stash information via misc (%m).
+	zstyle ':vcs_info:git*' formats '%b' '%R' '%a' '%m'
+	zstyle ':vcs_info:git*' actionformats '%b' '%R' '%a' '%m'
+	zstyle ':vcs_info:git*+set-message:*' hooks git-stash
```

Comment:
Ideally I’d like us to avoid “doing the work” for gut stash, unless it’s enabled. But it’s not critical for this PR and we cam change it afterwards.